tests/transactional/check_phub.pm

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Summary: Test packagehub repo on sle micro
+#
+# 1. Check if PackageHub extention is availble in sle micro
+# 2. Activate PackageHub extention
+# 3. Install a package from PackageHub repo
+#
+# Maintainer: qac <qa-c@suse.de>
+
+package check_phub;
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use transactional;
+
+sub run {
+    select_console 'root-console';
+    my $out = script_output("transactional-update --quiet register --list-extensions|grep PackageHub", proceed_on_failure => 1);
+    if ($out ne "") {    #check if PackageHub extension is available
+        if ($out =~ /Activate with: transactional-update /) {    #Check if it is activated
+            trup_call("$'");
```

Comment:
If you wish to use perlish vars, I guess this is going to be a better choice.
https://perldoc.perl.org/perlvar#$%7B%5EPOSTMATCH%7D
Firstly, it is self-explanatory and it doest no have the performance issue mentioned in https://perldoc.perl.org/perlvar#Performance-issues